### PR TITLE
AKS OSSKU supports Windows options

### PR DESCRIPTION
**Pull request by ShiqianTao**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Note
Refer to preview PR: #18706

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [ ] new API version
      - [x] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.
  2022 September.
  4. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month.
  As soon as possible.
  6. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [ ] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by ShiqianTao** [comment]
Could you help to review this PR? Thank you @AbelHu @FumingZhang

**Review by AbelHu — APPROVED** [review]

**Review by FumingZhang — APPROVED** [review]
LGTM

**Comment by FumingZhang** [comment]
See previous PR [#18706](https://github.com/Azure/azure-rest-api-specs/pull/18706) in preview API 2022-04-02-preview.

**Comment by ShiqianTao** [comment]
Could you help to review this PR? Thank you @palma21 @matthchr

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-07-01/managedClusters.json [hunk 1]
```diff
@@ -4597,13 +4600,15 @@
...
+        "CBLMariner",
+        "Windows2019",
+        "Windows2022"
       ],
       "x-ms-enum": {
         "name": "OSSKU",
         "modelAsString": true
       },
-      "description": "Specifies an OS SKU. This value must not be specified if OSType is Windows."
+      "description": "Specifies the OS SKU used by the agent pool. If not specified, the default is Ubuntu if OSType=Linux or Windows2019 if OSType=Windows. And the default Windows OSSKU will be changed to Windows2022 after Windows2019 is deprecated."
```
Comment (matthchr) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-07-01/managedClusters.json:4611]: Minor: The last sentence I think should just be: "The default Windows OSSKU will be changed to Windows2022 after Windows2019 is deprecated."

**Review by matthchr — APPROVED** [review]

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-07-01/managedClusters.json (hunk 1 above)
Comment (matthchr) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-07-01/managedClusters.json:4611]: By the way, do we backfill this property for old clusters? Say I created a Windows node pool already, and now I am using the new API. What value do I see when I do a GET on my ManagedCluster with the new API? I should see `Windows2019` right?

Comment (ShiqianTao) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2022-07-01/managedClusters.json]: Thank you @matthchr .
1. As discussed ```The default is Windows2022 when Kubernetes >= 1.25```, so add this information into the description directly. Is it suitable?
2. Yes. We have the plan that Regional Looper backfills OSSKU for Windows agent pools.

**Review by AbelHu — APPROVED** [review]

**Comment by ShiqianTao** [comment]
@leni-msft Could you help to merge this PR? Thank you.

**Pull request merged**